### PR TITLE
Fix pre-existing HintCode.UNNECESSARY_TYPE_CHECK_TRUE

### PR DESCRIPTION
**Pull request by scheglov**
This prepares for landing https://dart-review.googlesource.com/c/sdk/+/190360 into the analyzer.

### CODE REVIEW

**Review by kevmoo — APPROVED** [review]

**Review by jakemac53 — APPROVED** [review]

**Comment by jakemac53** [comment]
We should also bump the version to the next `-dev` and add a changelog entry for that version (can be empty).

**Comment by scheglov** [comment]
@jakemac53 I updated the version. Does it look right to you?

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,5 @@
+## 1.17.0-dev
```
Comment (jakemac53) [inline CHANGELOG.md:1]: Technically `1.16.1-dev` would be better (there are no new features, its just a patch release).

Comment (scheglov) [reply CHANGELOG.md]: Makes sense, updated.

**Review by jakemac53 — APPROVED** [review]

**Pull request merged**